### ISSUE TITLE
Not sure what kind of background layer is required for the plugin to use

### ISSUE THREAD

**Issue by joshlemer** [mthh/QgisDistanceCartogramPlugin#6]
Hey there, thanks for this plugin! I'm trying to use it to generate an image, and I'm really new to QGIS and GIS in general, so I'm not sure what kind of background layer is required for the plugin to use. I've tried adding an OpenStreetMap layer, and I've also tried using a raster layer (a screenshot of the area I want to deform). In both cases, the layers do not appear in the DistanceCartogram dialog's "background layer" dropdown. Could you tell me what kinds of layers are acceptable?

**Comment by mthh** [comment]
The background layer to be distorted must be a vector layer of Polygons/MultiPolygons (this is the condition for the layers to appear in the dropdown of the background layers).

In the example that produced the image in the README: 
- the point layer is the French prefectures (i.e. the department capitals)
- the background layer is the French territory (more precisely the French departments layer)
- the matrix contains the travel-time by road between these department capitals.

If you are new to GIS and are not familiar with geospatial data providers to find an appropriate background layer, you may want to check out this resource for a list of geospatial datasets: http://freegisdata.rtwilson.com/

I haven't used `QgisDistanceCartogramPlugin` recently so I hope there are no bugs (with recent versions of QGIS) in the version that is downloadable from QGIS plugins.
Once you have found a suitable background layer, please let me know if you encounter any other problems.

**Comment by mthh** [comment]
Furthermore, in the QGIS interface if you go to "Vector" > "DistanceCartogram" you can add an example dataset. This is the dataset I described in my previous message (look at the "prefecture" layer which is a Polygon/MultiPolygon vector layer).

**Comment by joshlemer** [comment]
Thanks @mthh! I will look at that website, for now I am looking at the sample dataset and an other issue I face is that the "help" documentation talks about performing a distortion without a "Reference Feature" but I don't see that this is possible through the UI. Both with my own dataset and with the sample dataset, the Reference Feature dropdown has no option for "none" and automatically chooses one of the features when the point layer is selected. 

![image](https://user-images.githubusercontent.com/5122903/205129358-b6d80336-1867-479e-8652-d47fdaeaad0d.png)

The other issue I am always having and I am really confused by is that my layers are apparently not a "projected CRS" and then I get these error messages:

![image](https://user-images.githubusercontent.com/5122903/205129930-d2ef0e9b-d4f5-4580-8012-688f5f520a91.png)

All I did was extract an OpenStreetMap road layer using the QuickOSM plugin (https://plugins.qgis.org/plugins/QuickOSM/) and it came out with these properties, so I'm not sure how to make the layer a projected layer
![image](https://user-images.githubusercontent.com/5122903/205131121-dcbd6e3d-c3b2-45b2-b3ca-3c872f36d01d.png)

**Comment by joshlemer** [comment]
Oh sorry in my example I showed my datasets have projection of OGC:CR84, but actually they are EPSG:4326, I'm not sure if that makes a difference. 
![image](https://user-images.githubusercontent.com/5122903/205132172-fe3ed10e-b15c-4a40-a556-966e5eceb664.png)

**Comment by joshlemer** [comment]
Oh okay I think I was able to figure out the CRS issue but now I get this traceback when I run the plugin 

```
2022-12-01T10:43:48     CRITICAL    Traceback (most recent call last):
              File "/Users/josh/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/dist_cartogram/worker.py", line 150, in run
              transformed_layers = self.get_transformed_layers()
              File "/Users/josh/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/dist_cartogram/worker.py", line 88, in get_transformed_layers
              new_geom = QgsGeometry.fromPolyLineXY([
             AttributeError: type object 'QgsGeometry' has no attribute 'fromPolyLineXY'
```

**Comment by joshlemer** [comment]
Well it seems that that issue was because I was using the road layer which is lines, but it only works with polygons maybe, so I tried it on a buildings layer. It worked well but that is not a great replacement so I will have to think of how to turn the road lines layer into polygons, thank you again for the great plugin!

**Comment by mthh** [comment]
I made some minor bug fixes and published a new version (v0.3) on the QGIS Plugin Repository.
In my tests I also managed to successfully transform not only Polygon layers but also LineString layers (and I couldn't reproduce the error message from your last message).
Please feel free to try with the latest version and let me know if you still have problems (and maybe if you can provide me with data to reproduce a problem, it could help me to debug).

**Comment by joshlemer** [comment]
Hey @mthh sorry I forgot to reply, yes the new version seems to work for the usecase, thank you!
